python/cuml/metrics/homogeneity_score.pyx

```diff
@@ -0,0 +1,90 @@
+#
+# Copyright (c) 2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+import numpy as np
+
+from libc.stdint cimport uintptr_t
+
+from cuml.common.handle cimport cumlHandle
+from cuml.utils import input_to_dev_array
+import cuml.common.handle
+cimport cuml.common.cuda
+
+cdef extern from "cuml/metrics/metrics.hpp" namespace "ML::Metrics":
+    double homogeneityScore(const cumlHandle & handle, const int *y,
+                            const int *y_hat, const int n,
+                            const int lower_class_range,
+                            const int upper_class_range) except +
+
+
+def homogeneity_score(ground_truth, predictions, handle=None):
+    """
+    Computes the homogeneity metric of a cluster labeling given a ground truth.
+
+    A clustering result satisfies homogeneity if all of its clusters contain
+    only data points which are members of a single class.
+
+    This metric is independent of the absolute values of the labels:
+    a permutation of the class or cluster label values won’t change the score
+    value in any way.
+
+        Parameters
+        ----------
+        handle : cuml.Handle
+        predictions : NumPy ndarray or Numba device
```

Comment:
This accepts numpy, numba device arrays, or cudf Series (since you're using input_to_dev_array_already)

Reply:
should specify that int32 is expected here

Reply:
Makes sense, thanks!